patches/v8/0003-Allow-Windows-builds-under-Bazel.patch

```diff
@@ -1,113 +1,9 @@
 
 
-diff --git a/BUILD.bazel b/BUILD.bazel
-index 6467518686d6e886c4f3a7f787551bc3be914462..c056e87386233c7d8e512dfa53f4108c8b42678e 100644
---- a/BUILD.bazel
-+++ b/BUILD.bazel
-@@ -822,6 +822,7 @@ filegroup(
-         "src/base/platform/mutex.h",
-         "src/base/platform/platform.cc",
-         "src/base/platform/platform.h",
-+        "src/base/platform/platform-posix.h",  # Always included by src/execution/isolate.h
-         "src/base/platform/semaphore.cc",
-         "src/base/platform/semaphore.h",
-         "src/base/platform/time.cc",
-@@ -866,7 +867,6 @@ filegroup(
-     ] + select({
-         "@v8//bazel/config:is_posix": [
-             "src/base/platform/platform-posix.cc",
--            "src/base/platform/platform-posix.h",
-             "src/base/platform/platform-posix-time.cc",
-             "src/base/platform/platform-posix-time.h",
-         ],
-@@ -889,6 +889,7 @@ filegroup(
-         "@v8//bazel/config:is_windows": [
-             "src/base/debug/stack_trace_win.cc",
-             "src/base/platform/platform-win32.cc",
-+            "src/base/platform/platform-win32.h",
-             "src/base/win32-headers.h",
-         ],
-     }),
-@@ -1271,6 +1272,7 @@ filegroup(
-         "include/v8-wasm-trap-handler-posix.h",
-         "src/api/api.cc",
-         "src/api/api.h",
-+        "include/v8-wasm-trap-handler-win.h",
-         "src/api/api-arguments.cc",
-         "src/api/api-arguments.h",
-         "src/api/api-arguments-inl.h",
-@@ -2757,6 +2759,11 @@ filegroup(
-             "src/trap-handler/handler-inside-posix.cc",
-             "src/trap-handler/handler-outside-posix.cc",
-         ],
-+        "@v8//bazel/config:is_windows": [
-+            "src/trap-handler/handler-inside-win.cc",
-+            "src/trap-handler/handler-inside-win.h",
-+            "src/trap-handler/handler-outside-win.cc",
-+        ],
-         "//conditions:default": [],
-     }) + select({
-         "@v8//bazel/config:v8_arm64_simulator": [
-@@ -2764,13 +2771,6 @@ filegroup(
-             "src/trap-handler/trap-handler-simulator.h",
-         ],
-         "//conditions:default": [],
--    }) + select({
--        "@v8//bazel/config:is_windows": [
--            "src/trap-handler/handler-inside-win.cc",
--            "src/trap-handler/handler-inside-win.h",
--            "src/trap-handler/handler-outside-win.cc",
--        ],
--        "//conditions:default": [],
-     }) + select({
-         "@v8//bazel/config:is_windows_64bit": [
-             "src/diagnostics/unwinding-info-win64.cc",
-@@ -3833,6 +3833,9 @@ filegroup(
-         "@v8//bazel/config:is_msvc_asm_ia32": ["src/heap/base/asm/ia32/push_registers_masm.asm"],
-         "@v8//bazel/config:is_msvc_asm_x64": ["src/heap/base/asm/x64/push_registers_masm.asm"],
-         "@v8//bazel/config:is_msvc_asm_arm64": ["src/heap/base/asm/arm64/push_registers_masm.S"],
-+        "@v8//bazel/config:is_windows_clang_asm_ia32": ["src/heap/base/asm/ia32/push_registers_asm.cc"],
```

Comment:
I was able to upstream parts of this patch, but didn't try to land potentially controversial aspects like clang-cl support – we still need this part of the patch.